### PR TITLE
[rush-lib] Extend LookupByPath to return index of remaining path

### PR DESCRIPTION
**Pull request by dmichon-msft**
## Summary
Enhances the `LookupByPath` helper class with a method `findChildPathAndIndex` that returns both the matched entry and the index at which the remainder of the query path starts.

## Details
Useful for `ProjectChangeAnalyzer` when routing files to projects, to get the remaining path relative to the project.

## How it was tested
Added unit tests.

## Impacted documentation
None

### CODE REVIEW

File: libraries/rush-lib/src/logic/LookupByPath.ts [hunk 1]
```diff
@@ -146,6 +170,23 @@ export class LookupByPath<TItem> {
...
+   * @returns the found item, or `undefined` if no item was found
+   *
+   * @example
+   * ```ts
+   * const tree = new LookupByPath([['foo', 1], ['foo/bar', 2]]);
+   * tree.findChildPathAndIndex('foo/baz'); // returns { item: 1, index: 4 }
+   * tree.findChildPathAndIndex('foo/bar/baz'); // returns { item: 2, index: 8 }
+   * ```
+   */
+  public findChildPathAndIndex(childPath: string): IPrefixMatch<TItem> | undefined {
```
Comment (D4N14L) [inline libraries/rush-lib/src/logic/LookupByPath.ts:186]:
```suggestion
  public findNearestChildPathAndIndex(childPath: string): IPrefixMatch<TItem> | undefined {
```

**Review by D4N14L — APPROVED** [review]

File: libraries/rush-lib/src/logic/LookupByPath.ts (hunk 1 above)
Comment (dmichon-msft) [reply libraries/rush-lib/src/logic/LookupByPath.ts:186]: Is this a request for a general renaming of the API surface, or just this specific new method?

Comment (D4N14L) [reply libraries/rush-lib/src/logic/LookupByPath.ts:186]: General renaming of the API surface. This is a new API, just feel that the name suggested fits what it does better than the name it has is all

Comment (dmichon-msft) [reply libraries/rush-lib/src/logic/LookupByPath.ts]: Not renaming the existing API for now since some external code is known to take a dependency.

**Review by D4N14L — APPROVED** [review]

**Pull request merged**